src/components/entity/ha-state-label-badge.js

```diff
@@ -78,6 +81,12 @@ export default new Polymer({
       }
       // state == 'disarmed'
       return 'mdi:lock-open';
+    case 'binary_sensor':
```

Comment:
It can join `device_tracker`, `scene` and `script` and delegate icon to `domainIcon` (see other comment). While at it, could you add `updater` to the list? No need for it's own call to domainIcon.